lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -266,26 +302,49 @@ if Enum.all?(
         }
       end
     else
-      defp hls_links_and_children(_link_builder, :OPUS, _track, _segment_duration, _framerate) do
+      defp hls_links_and_children(
+             _link_builder,
+             :OPUS,
+             _track,
+             _segment_duration,
+             _framerate,
+             _transcoding_config
+           ) do
         raise """
         Cannot find one of the modules required to support Opus audio input.
         Ensure `:membrane_opus_plugin`, `:membrane_aac_plugin` and `:membrane_aac_fdk_plugin` are added to the deps.
         """
       end
     end
 
-    defp hls_links_and_children(link_builder, :AAC, track, _segment_duration, _framerate),
-      do: %ParentSpec{
-        children: %{},
-        links: [
-          link_builder
-          |> via_in(Pad.ref(:input, {:audio, track.id}), options: [encoding: :AAC])
-          |> to({:hls_sink_bin, track.stream_id})
-        ]
-      }
-
-    defp hls_links_and_children(link_builder, :H264, track, segment_duration, framerate),
-      do: %ParentSpec{
+    defp hls_links_and_children(
+           link_builder,
+           :AAC,
+           track,
+           _segment_duration,
+           _framerate,
+           _transcoding_config
+         ),
+         do: %ParentSpec{
+           children: %{},
+           links: [
+             link_builder
+             |> via_in(Pad.ref(:input, {:audio, track.id}), options: [encoding: :AAC])
+             |> to({:hls_sink_bin, track.stream_id})
+           ]
+         }
+
+    defp hls_links_and_children(
+           link_builder,
+           :H264,
+           track,
+           segment_duration,
+           framerate,
+           transcoding_config
+         ) do
+      transcoding_interceptor = create_transcoding_interceptor(transcoding_config, track.id)
```

Comment:
This indeed can be thought of as an interceptor but I think that `link_to_transcoder = create_transcoder/create_transcoder_link` will fit better, we will have 

```
          |> to({:keyframe_requester, track.id})
          |> to({:video_parser, track.id})
          |> then(link_to_transcoder)
```